addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -715,10 +715,8 @@ def _do_create_moves(self):
         for sheet in own_account_sheets:
             sheet.accounting_date = sheet.accounting_date or sheet._calculate_default_accounting_date()
         moves_sudo = self.env['account.move'].sudo().create([sheet._prepare_bills_vals() for sheet in own_account_sheets])
-        # Set the main attachment on the moves directly to avoid recomputing the
-        # `register_as_main_attachment` on the moves which triggers the OCR again
         for move_sudo in moves_sudo:
-            move_sudo.message_main_attachment_id = move_sudo.attachment_ids[0] if move_sudo.attachment_ids else None
+            move_sudo._message_set_main_attachment_id(move_sudo.attachment_ids.ids, force=True)
```

Comment:
Note that before it was using the first attachment regardless of XML type and any sorting of any kind.

I'm not sure it was fully intended though.